File: ios/YLNote/最新笔记/YLPerNoteViewController.m
//
//  YLPerNoteViewController.m
//  YLNote
//
//  Created by tangh on 2021/6/28.
//  Copyright © 2021 tangh. All rights reserved.
//

#import "YLPerNoteViewController.h"
#import "YLNote-Swift.h"
#import <Masonry/Masonry.h>

@interface YLPerNoteViewController ()<YLQuestionDataProtocol>

@property (nonatomic,strong)UITableView *table;

@property (nonatomic,strong) YLPerNoteQesManager *dataManager;
@property (nonatomic,copy) NSArray<YLNoteGroup *> *sectionDatas;


@end

@implementation YLPerNoteViewController

- (void)setupUI {
    self.title = @"笔记整理";
    self.sectionDatas = self.dataManager.allDatas;
    self.table.delegate = self.dataManager;
    self.table.dataSource = self.dataManager;
    self.dataManager.funcHandler = ^(YLQuestion * question, id prama) {
        NSLog(@"%@",question);
    };

    [self.table registerClass:[YLGroupHeaderView class] forHeaderFooterViewReuseIdentifier:[self.dataManager headerIdentifier]];
    
    NSString *path = [[NSBundle mainBundle] pathForResource:@"YLQuestionTableViewCell" ofType:@"nib"];
    if (path) {
        [self.table registerNib:[UINib nibWithNibName:@"YLQuestionTableViewCell" bundle:nil] forCellReuseIdentifier:[self.dataManager cellIdentifier]];
    } else {
        [self.table registerClass:[UITableViewCell class] forCellReuseIdentifier:[self.dataManager cellIdentifier]];
    }
}


- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    [self setupUI];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}
#pragma mark - 点击分组信息
//- (void)clickGroupAction:(UIButton *)sender {
//    YLNoteGroup *sectionData = self.sectionDatas[sender.tag - 1000];
//    BOOL groupStatus = sectionData.unfoldStatus;
//    if(!groupStatus){
//        sectionData.unfoldStatus = YES;
//    } else {
//        sectionData.unfoldStatus = NO;
//    }
//    //刷新当前的分组
//    NSIndexSet * set=[[NSIndexSet alloc] initWithIndex:sender.tag - 1000];
//    [self.table reloadSections:set withRowAnimation:UITableViewRowAnimationNone];
//}


#pragma mark - datasource Protocol
//- (NSString *)jsonFileName {
//    return @"MyNote";
//}
//
//- (NSString *)cellIdentifier {
//    return @"kMyNote_cell";
//}
//
//- (NSString *)headerIdentifier {
//    return @"kYLGroupHeaderView";
//}

//- (void)doFunctionWith:(NSString *)name parameter:(id)parameter {
//    NSString *funcName = [NSString stringWithFormat:@"test_%@",name];
//    SEL selector = NSSelectorFromString(funcName);
//    
//    //检查是否有"myMethod"这个名称的方法
//    if ([self respondsToSelector:selector]) {
//        if (!self) { return; }
//        if ([name containsString:@":"]) {
//            IMP imp = [self methodForSelector:selector];
//            CGRect (*func)(id, SEL, id, id) = (void *)imp;
//            CGRect result = self ?
//            func(self, selector, @"2", @"3") : CGRectZero;
//            NSLog(@"result:: %@",NSStringFromCGRect(result));
//        } else {
//            IMP imp = [self methodForSelector:selector];
//            void (*func)(id, SEL) = (void *)imp;
//            func(self, selector);
//        }
//    }
//    
//    
//}

#pragma mark - lazy

- (UITableView *)table {
    if (!_table) {
        _table = [[UITableView alloc] initWithFrame:CGRectZero style:UITableViewStyleGrouped];
        [self.view addSubview:self.table];
        [_table mas_makeConstraints:^(MASConstraintMaker *make) {
            make.edges.mas_equalTo(0);
        }];
    }
    return _table;
}

- (YLPerNoteQesManager *)dataManager {
    if (!_dataManager) {
        _dataManager = [[YLPerNoteQesManager alloc] init];
    }
    return _dataManager;
}

@end

File: ios/YLNote/Tab-VCs/最新笔记/YLPerNoteViewController.swift
//
//  YLPerNoteViewController.swift
//  YLNote
//
//  Created by tangh on 2022/7/19.
//  Copyright © 2022 tangh. All rights reserved.
//

import UIKit

class YLPerNoteQesManager: YLQuestionDataManagerBase {
    override var jsonFileName: String {
        return "PersonNote";
    }
    
    override var cellIdentifier: String {
        return "YLPerNoteViewController.cell";
    }
    
    override var headerIdentifier: String {
        return "YLPerNoteViewController.header";
    }
}

class YLPerNoteViewController: UIViewController {
    deinit {
        print("\(self.description): \(#function)")
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = #colorLiteral(red: 0.9764705896, green: 0.850980401, blue: 0.5490196347, alpha: 1)
        self.navigationItem.title = "笔记整理"
        setupUI()
    }
    
    func setupUI() {
        view.addSubview(self.table)
        self.table.dataSource = self.dataManager;
        self.table.delegate = self.dataManager;
        self.dataManager.funcHandler = { (ques,pram) in
            print("\(ques.description)")
        }
        
        let name = String(describing: YLQuestionTableViewCell.self)
        table.register(UINib(nibName: name, bundle: nil), forCellReuseIdentifier: dataManager.cellIdentifier)
        table.register(YLGroupHeaderView.self, forHeaderFooterViewReuseIdentifier: dataManager.headerIdentifier)
    }
    
    //MARK: lazy method
    lazy var table: UITableView = {
        let t = UITableView(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height), style: .grouped )
        t.rowHeight = 40
        
        return t
    }()
    
    lazy var dataManager: YLPerNoteQesManager = {
        let manager = YLPerNoteQesManager()
        return manager
    }()
    
    var sectionDatas: [YLNoteGroup] {
        return self.dataManager.allDatas
    }
    
}

